### ISSUE TITLE
plot_surface gives blank figure with log scale for axes

### ISSUE THREAD

**Issue by stevengj** [matplotlib/matplotlib#9545]
Setting the x or y axes to a log scale seems to result in a blank figure with `plot_surface`.  For example:

```py
import numpy as	np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D

x = np.logspace(-3,1,20)[np.newaxis].T
y = np.logspace(-1,2,30)
z = x**2 + y**3 # create surface data via broadcasting

fig = plt.figure()
ax = fig.gca(projection='3d')
ax.plot_surface(x,y,z)

# works fine if we comment out these lines:
ax.set_xscale("log")
ax.set_yscale("log")

plt.show()
```
Commenting out the `set_xscale` and `set_yscale` lines results in a working (visible) plot, but with those lines I get an empty plot window.

Since the `x` and `y` coordinates are all nonnegative, I would have expected a working log scale.

**Matplotlib version**
  * Operating system: MacOS 10.11.6
  * Matplotlib version: 2.1.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): MacOSX
  * Python version: Python 2.7.14 (installed via Anaconda)

**Comment by WeatherGod** [comment]
mplot3d does not work properly with non-linear axes scales. This has been a
long-running issue due to the fact that mplot3d predates much of
matplotlib's transforms system (which includes support for things like log
scales).

On Mon, Oct 23, 2017 at 1:01 PM, Steven G. Johnson <notifications@github.com

**Comment by stevengj** [comment]
I see.  Would be nice if it could throw an informative exception in the meantime.

**Comment by ImportanceOfBeingErnest** [comment]
I guess this can be merged into the old #209

**Comment by anntzer** [comment]
Indeed.
